### DISCUSSION TITLE
Instructional Guide; DS3231 Real-Time Clock; micro:bit v2; micropython v3

### DISCUSSION THREAD

**Discussion by burksfamly** [micropython/micropython#14389] [nRF]
I have attached a zipped instructional guide for using the DS3231 real-time clock with the 
micro:bit v2 and micropython v3 development environment. The guide covers many topics
and instructional objectives for beginners including:

- how to layout, create, and use a very simple class structure
- how to use the I2C bus for I2C devices connected to the micro:bit
- how to use a microchip datasheet and communicate with the microchip
- bitwise logical operations and byte and bit manipulation
- binary-coded decimals and two's compliment encoding of byte structures
- memoryview and data manipulation
- how to read and write data bytes to/from microchip registers
- how to exploit the features of the ds3231 RTC

The zipped file is linked below:

[DS3231 RTC microbit micropython guide.zip](https://github.com/micropython/micropython/files/15143128/DS3231.RTC.microbit.micropython.guide.zip)

**Comment by peterhinch** [comment]
See also [this doc](https://github.com/peterhinch/micropython-samples/tree/master/DS3231).
